src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1107,29 +1029,39 @@ public TValue GetOrAdd(TKey key, TValue value)
         /// <param name="updateValueFactory">The function used to generate a new value for an existing key
         /// based on the key's existing value</param>
         /// <param name="factoryArgument">An argument to pass into <paramref name="addValueFactory"/> and <paramref name="updateValueFactory"/>.</param>
-        /// <exception cref="System.ArgumentNullException"><paramref name="key"/> is a null reference
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is a null reference
         /// (Nothing in Visual Basic).</exception>
-        /// <exception cref="System.ArgumentNullException"><paramref name="addValueFactory"/> is a null reference
+        /// <exception cref="ArgumentNullException"><paramref name="addValueFactory"/> is a null reference
         /// (Nothing in Visual Basic).</exception>
-        /// <exception cref="System.ArgumentNullException"><paramref name="updateValueFactory"/> is a null reference
+        /// <exception cref="ArgumentNullException"><paramref name="updateValueFactory"/> is a null reference
         /// (Nothing in Visual Basic).</exception>
-        /// <exception cref="System.OverflowException">The dictionary contains too many
+        /// <exception cref="OverflowException">The dictionary contains too many
         /// elements.</exception>
         /// <returns>The new value for the key.  This will be either be the result of addValueFactory (if the key was
         /// absent) or the result of updateValueFactory (if the key was present).</returns>
         public TValue AddOrUpdate<TArg>(
             TKey key, Func<TKey, TArg, TValue> addValueFactory, Func<TKey, TValue, TArg, TValue> updateValueFactory, TArg factoryArgument)
         {
-            if (key == null) ThrowKeyNullException();
-            if (addValueFactory == null) throw new ArgumentNullException(nameof(addValueFactory));
-            if (updateValueFactory == null) throw new ArgumentNullException(nameof(updateValueFactory));
+            if (key is null)
+            {
+                ThrowHelper.ThrowKeyNullException();
+            }
+
+            if (addValueFactory is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(addValueFactory));
+            }
+
+            if (updateValueFactory is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(updateValueFactory));
+            }
```

Comment:
```suggestion
            if (addValueFactory is null || updateValueFactory is null)
            {
                ThrowHelper.ThrowArgumentNullException(
                    addValueFactory == null ? nameof(addValueFactory) : nameof(updateValueFactory));
            }
```

[shorter code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8lAbhvoYGUALbKAA4AZbMAB0AJQCuAOwwBLfDHbUOAZiakGAYQYBvGgyNMNxFAwAaACmaNsAE3sA1bABspMAGLYwGaAE80JnJGKQF7bAwYF3cvHz8ofwBKBkNjA2pjLOz5ADMGKwdnNw9vXwCGeVwGGSlXVyS07OyM5rbjABUeKAgAdwAJGFcBGCgxLp7egEEoAHMpJTkAOTrXAFEEMBgBBQgZKxlsJQhcwscY0viApKSVdqyAXxom+7yCsIioi7jyxMrq2r1RqZe5GVqg7ITPqDYajcbdPozeaLDAreobLY7eR7A5HGAnKwfSLREo/BLJW4vdpPEFGGlZF7EUzmCykGwhBhFb5lclBWwMIlfUk866pWn6KltN5nYqxEV/Ko1VYMQiEAXhYncq4KgGrYEQiXig1QgZDEZjE1IhYwZarDHbXb7SUGoxc4Xa/wMAC8XqV9QYAH4aniCW65R6UiBg8dToKSeHfhS7hD6cYaTT1JoGCaYeaaODjEzgkgmOZLXNrbb0ZsHdj9vz+LN9VkC80MAjejUYJ2rSi0esa1icY3KUb00A==)?

Reply:
> shorter code?

Try adding anything after your last if block, like a Console.WriteLine().